enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,893 @@
+is completed the `bundle-laoder` will update the new
+`status.desired.bundle.nodes.*.loaded` field. For example, when the images have
+been loaded for `node0` the status will look like this:
+
+```yaml
+status:
+  desired:
+    bundle:
+      monitor: true
+      file: /dev/sdb
+      nodes:
+        node0:
+          extracted: true
+          loaded: true
+          metadata: '{ ... }'
+        node1:
+          extracted: true
+          loaded: false
+          metadata: '{ ... }'
+        node2:
+          extracted: true
+          loaded: false
+          metadata: '{ ... }'
+```
+
+At that point the `bundle-loader` job will finish.
+
+When all the nodes have the bundle loaded (the `loaded` fields of all nodes is
+`true`) the cluster version operator will check the new
+`spec.desiredUpdate.hold` field of the `ClusterVersion` object. This field will
+be used to indicate if the upgrade has to be started immediately, or if should
+wait till the user explicitly sets it to `false`. The default value for that
+will be `false`. This is intended for situations where the user wishes to
+prepare everything for the upgrade in advance, but wants to perform the upgrade
+later.
+
+When the value of the `spec.desiredUpdate.hold` field is `false` the cluster
+version operator will trigger the regular upgrade process, adding something
+like this to the `ClusterVersion` object:
+
+```yaml
+spec:
+  desiredUpdate:
+    image: quay.io/openshift-release-dev/ocp-release@...
+```
+
+The value of the `image` field will be obtained from the `release` field of the
+bundle metadata.
+
+When the upgrade has completed successfully the cluster version operator will
+delete the `ContainerRuntimeConfig` object that it created to pin the images, and
+will start a new `bundle-cleaner` job in each node that will clean all the
+artifacts potentially left around by other pieces of the upgrade. In particular
+it will remove the `/var/lib/upgrade/4.13.7-x86_64` directory created by the
+`bundle-extractor`.
+
+In order to support the situation where the user only wants to load the images
+in advance but doesn't want to use a bundle, the `ClusterVersion` spec will have
+a `desiredUpdate.bundle.enabled` field. The default will be `true`, but when
+explicitly set to false the the cluster version operator will skip the creation
+of the `bundle-server` and the `bundle-extractor` and will go directly to create
+the `bundle-loader`. The `bundle-loader` will in this case skip the step to
+start the embedded registry and configure it as a mirror. Instead of that it will
+ask CRI-O to pull the images as usual.
+
+
+### Risks and Mitigations
+
+The proposed solution will require space to store the release bundle and all the
+release images in all the nodes of the cluster, approximately 48 GiB in the
+worst case. To mitigate that risks the components that will consume disk space
+will check in advance if the required space is available.
```

Comment:
The storage requirement also includes the images for the running cluster, right?

Reply:
Those 48 GiB include 16 GiB required to extract the contents of the bundle to a temporary directory and the 32 GiB that the release images will take in the `/var/lib/containers/storage` directory once copied. It doesn't include the space required for the already existing images.

Note that the 16 GiB used to temporarily extract the bundle are released once the images are copied.